### PR TITLE
Controlfreec

### PR DESCRIPTION
**Pull request by MaxUlysse**
- Adding `samtools mpileup`
- Adding `Control-FREEC`

## PR checklist
 - [X] PR is made against `dev` branch
 - [X] This comment contains a description of changes (with reason)
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [X] Ensure the test suite passes (`./scripts/test.sh -p docker -t ALL`).
 - [ ] Documentation in `docs` is updated
 - [ ] `CHANGELOG.md` is updated
 - [ ] `README.md` is updated

### CODE REVIEW

**Comment by MaxUlysse** [comment]
squash and merge when merging, don't want to have that many `Merge remote-tracking branch 'upstream/master' into controlfreec`

File: conf/containers.config [hunk 1]
```diff
@@ -8,120 +8,48 @@
  */
 
 process {
```
Comment (MaxUlysse) [inline conf/containers.config]: I think this can definitively be improved ;-)

File: conf/genomes.config [hunk 2]
```diff
@@ -52,6 +52,8 @@ params {
       acLoci           = "${params.genome_base}/1000G_phase3_20130502_SNP_maf0.3.small.loci"
       acLociGC         = "${params.genome_base}/1000G_phase3_20130502_SNP_maf0.3.small.loci.gc"
       bwaIndex         = "${params.genome_base}/human_g1k_v37_decoy.small.fasta.{amb,ann,bwt,pac,sa}"
+      chrDir           = "${params.genome_base}/chr/"
```
Comment (MaxUlysse) [inline conf/genomes.config]: Will need to add that to iGenomes

File: conf/genomes.config [hunk 3]
```diff
@@ -52,6 +52,8 @@ params {
...
+      chrLength        = "${params.genome_base}/human_g1k_v37_decoy.small.len"
```
Comment (MaxUlysse) [inline conf/genomes.config]: Will need to add that to iGenomes

File: conf/igenomes.config [hunk 4]
```diff
@@ -19,6 +19,7 @@ params {
       acLoci           = "${params.genome_base}/Annotation/ASCAT/1000G_phase3_20130502_SNP_maf0.3.loci"
       acLociGC         = "${params.genome_base}/Annotation/ASCAT/1000G_phase3_20130502_SNP_maf0.3.loci.gc"
       bwaIndex         = "${params.genome_base}/Sequence/BWAIndex/human_g1k_v37_decoy.fasta.{amb,ann,bwt,pac,sa}"
+      chrDir           = "${params.genome_base}/Sequence/Chromosomes/"
```
Comment (MaxUlysse) [inline conf/igenomes.config]: Where is `chrLenght`?

File: conf/igenomes.config [hunk 5]
```diff
@@ -49,6 +52,7 @@ params {
       acLoci           = "${params.genome_base}/1000G_phase3_20130502_SNP_maf0.3.small.loci"
       acLociGC         = "${params.genome_base}/1000G_phase3_20130502_SNP_maf0.3.small.loci.gc"
       bwaIndex         = "${params.genome_base}/human_g1k_v37_decoy.small.fasta.{amb,ann,bwt,pac,sa}"
+      chrDir           = "${params.genome_base}/chr/"
```
Comment (MaxUlysse) [inline conf/igenomes.config]: Where is `chrLenght`?

File: docs/CONTAINERS.md [hunk 6]
```diff
@@ -11,27 +11,16 @@ For annotation, the main container can be used, but the cache has to be download
...
-
-### runallelecount [![runallelecount-docker status][runallelecount-docker-badge]][runallelecount-docker-link]
-
-- Based on `debian:8.9`
-- Contain **[AlleleCount][allelecount-link]** 2.2.0
 
 ### sarek [![sarek-docker status][sarek-docker-badge]][sarek-docker-link]
 
 - Based on `nfcore/base:latest`
+- Contain **[AlleleCount][allelecount-link]** 2.1.2
```
Comment (MaxUlysse) [inline docs/CONTAINERS.md]: Need to update Bioconda recipe for AlleleCount

File: environment.yml [hunk 7]
```diff
@@ -7,12 +7,19 @@ channels:
...
 
 dependencies:
+  # conda-forge packages
+  - r=3.5.1
+  - r-rcolorbrewer=1.1
+  # bioconda packages
   - bcftools=1.8
+  - bioconductor-rtracklayer=1.42.1
   - bwa=0.7.17
+  - cancerit-allelecount=2.1.2
```
Comment (MaxUlysse) [inline environment.yml]: Need to update Bioconda recipe.
We should at least have the 2.2.0

File: environment.yml [hunk 8]
```diff
@@ -7,12 +7,19 @@ channels:
...
+  - control-freec=11.4
   - ensembl-vep=95.1
   - fastqc=0.11.8
-  - font-ttf-dejavu-sans-mono=2.37 #for FastQC
-  - fontconfig=2.12.6 #for FastQC
+  - font-ttf-dejavu-sans-mono=2.37 # Needed for FastQC
```
Comment (MaxUlysse) [inline environment.yml]: That could be removed

File: environment.yml [hunk 9]
```diff
@@ -7,12 +7,19 @@ channels:
...
+  - fontconfig=2.13                # Needed for FastQC
```
Comment (MaxUlysse) [inline environment.yml]: That could be removed

File: conf/genomes.config (hunk 3 above)
Comment (MaxUlysse) [reply conf/genomes.config]: And to UPPMAX

File: environment.yml (hunk 7 above)
Comment (MaxUlysse) [reply environment.yml]: https://github.com/bioconda/bioconda-recipes/tree/master/recipes/cancerit-allelecount

File: conf/containers.config (hunk 1 above)
Comment (MaxUlysse) [reply conf/containers.config]: Will be done in #789

**Comment by MaxUlysse** [comment]
Sometime, one small issue, in the R scripts after `freec`
```
Error in scan(file = file, what = what, sep = sep, quote = quote, dec = dec,  :
    line 579 did not have 9 elements
  Calls: read.table -> scan
  Execution halted
```
Caused by fact that sometime one line in the output from `freec` doesn't have the right number of elements.

Possible solutions:
- Do nothing, and explain that in the docs.
- Make a small bash script to remove the faulty line in the `freec` output

File: conf/genomes.config (hunk 3 above)
Comment (apeltzer) [reply conf/genomes.config]: How do we do this without @ewels?

**Comment by MaxUlysse** [comment]
I'll add it to an S3 bucket and ask Phil to sync it whenever he has some time, in the meanwhile, it's a very small file, that can be shared easilly

**Pull request closed without merging**